src/ophyd_async/epics/adandor/_andor_io.py

```diff
@@ -16,23 +16,30 @@ class Andor2TriggerMode(StrictEnum):
     SOFTWARE = "Software"
 
 
-class ImageMode(StrictEnum):
+class Andor2ImageMode(StrictEnum):
     SINGLE = "Single"
     MULTIPLE = "Multiple"
     CONTINUOUS = "Continuous"
     FAST_KINETICS = "Fast Kinetics"
 
 
+class Andor2DataType(SubsetEnum):
+    UINT16 = "UInt16"
+    UINT32 = "UInt32"
+    FLOAT32 = "Float32"
+    FLOAT64 = "Float64"
+
+
 class Andor2DriverIO(ADBaseIO):
     """
     Epics pv for andor model:DU897_BV as deployed on p99
     """
 
     def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(prefix, name=name)
         self.trigger_mode = epics_signal_rw(Andor2TriggerMode, prefix + "TriggerMode")
         self.data_type = epics_signal_r(ADBaseDataType, prefix + "DataType_RBV")
         self.andor_accumulate_period = epics_signal_r(
             float, prefix + "AndorAccumulatePeriod_RBV"
         )
-        self.image_mode = epics_signal_rw_rbv(ImageMode, prefix + "ImageMode")
-        super().__init__(prefix, name=name)
+        self.image_mode = epics_signal_rw_rbv(Andor2ImageMode, prefix + "ImageMode")
```

Comment:
This seems a relatively common pattern, overriding the image mode to be a subset of what the baseclass provides. We have `SubsetEnum` already, should we have `SupersetEnum` too? Or we could drop to string in the baseclass...